packages/typespec-vscode/src/extension.ts

```diff
@@ -29,7 +29,7 @@ import { importFromOpenApi3 } from "./vscode-cmd/import-from-openapi3.js";
 import { installCompilerGlobally } from "./vscode-cmd/install-tsp-compiler.js";
 import { clearOpenApi3PreviewTempFolders, showOpenApi3 } from "./vscode-cmd/openapi3-preview.js";
 
-let client: TspLanguageClient | undefined;
+export let client: TspLanguageClient | undefined;
```

Comment:
it's better to pass client to emit-code as parameter instead of through global export to avoid circle dependency when possible

Reply:
move client into another file to resolve circle dependency.